pynxtools/dataconverter/readers/ellips/reader.py

```diff
@@ -466,6 +467,16 @@ def read(self,
         for index in range(1, len(data_list)):
             template["/ENTRY[entry]/plot/@auxiliary_signals"] += data_list[index]
 
+        template["/ENTRY[entry]/definition"] = "NXellipsometry"
+        template["/ENTRY[entry]/definition/@url"] = (
+            "https://github.com/FAIRmat-NFDI/nexus_definitions/"
+            "blob/fairmat/contributed_definitions/NXellipsometry.nxdl.xml"
+        )
+        template["/ENTRY[entry]/definition/@version"] = get_nexus_version()
```

Comment:
should not it be put into template generation rather then overwriting what the user may have specified?

Reply:
IMO this *should* overwrite. Even if the user specified it we use this version from pynxtool's submodule for creating the file.

Reply:
#174 is related to this. I will cross-link this so discussions can be followed in both places. If there is a clear decision here, we can close #174.

I don't have a strong opinion either side. But our project is designed to be used from PyPi. Therefore, overwriting the version to the one that was used to verify is the more coherent way forward. 

If we have an advanced user, they could easily bypass any checks and just use the Writer class with a Template object to write whatever they desired.

Reply:
Yes I would vote for this option from #174:


> The other suggestion is to give an error if the attrib provided by the user is different in value then what the default write function provides. This also makes sense for sensitive attributes like Application Definition or NeXus Version. But if the user really wants to override, we will need to provide a flag for that.